features/analytics/api/src/main/kotlin/io/element/android/features/analytics/api/preferences/AnalyticsPreferencesView.kt

```diff
@@ -43,6 +45,13 @@ fun AnalyticsPreferencesView(
     )
     Column(modifier) {
         ListItem(
+            modifier = Modifier
+                .toggleable(
+                    value = state.isEnabled,
+                    role = Role.Checkbox,
+                    enabled = true,
+                    onValueChange = { onEnabledChanged(!state.isEnabled) }
+                ),
```

Comment:
I am wondering if all these changes could be handle at a single place: in the `ListItem` implementation. It will reduce the number of change and will have the advantage of always use `toggleable`/`selectable` so that developer will not forget about it.

Here is a patch with a proposal:
[ListItemModifier.patch](https://github.com/user-attachments/files/18151809/ListItemModifier.patch)

Note: not tested and may not always work, this patch is just to give some guidance about how it could be implemented. More work is necessary to make this patch an acceptable pull request.

Reply:
Hello again,
The patch is working as it should with no modifications from me.
It correctly applies modifiers as expected.
However I still have to go through the places I have modified and verify if the list item is really used in a way this implementation is expecting it.
For example:
* If there is an onClick handler on the switch or check box it-self it's being presented twice with some screen readers. Having click handler on both list item and switch / checkbox should therefore be avoided if we wish to make the screen reader experience fluent. Currently at least the favourite item in the room list long click menu is implemented like this.
* Some radio buttons are not working and I need to find out why for example when going to settings -> Advanced, there is a setting that affects app theme, it can be System, Light or Dark. And once the dialog is showing the radio buttons don't currently have the modifier applied as expected.

So if you are okay with the outcome and you do have enough bandwidth for doing it, please consider pushing your patch, in favour of this pull request and we should handle edge cases as follow up PRs. It's great improvement and it enables screen reader users to adjust settings on their own if they wish.